lib/streamlit/runtime/state/session_state.py

```diff
@@ -229,7 +231,7 @@ def get_serialized(self, k: str) -> WidgetStateProto | None:
         if is_array_value_field_name(field):
             arr = getattr(widget, field)
             arr.data.extend(serialized)
-        elif field == "json_value":
+        elif field in {"json_value", "json_trigger_value"}:
```

Comment:
We have used `string_value` wherever we have JSONs in our native widgets. There was a reason that I don't fully remember anymore. I believe it had to do with the automatic serialization causing more issues than benefits. I need to think a bit more about this. Maybe that's what required you to add the `presenter,` which, in case of keeping it basic as `string_value,` can be handled just by the deserializer.